src/soc/ibm/power9/isteps/istep8.c

```diff
@@ -0,0 +1,278 @@
+	}
+	/* breakpoint ratio: core floor 2.5, pb 2.0 (cache floor :: pb = 5/8) */
+	else if ((4 * freq_core_floor) >= (5 * (*fbc_freq))) {
+		*floor_ratio = FABRIC_CORE_FLOOR_RATIO_RATIO_5_8;
+	}
+	/* breakpoint ratio: core floor 2.0, pb 2.0 (cache floor :: pb = 4/8) */
+	else if (freq_core_floor >= (*fbc_freq)) {
+		*floor_ratio = FABRIC_CORE_FLOOR_RATIO_RATIO_4_8;
+	}
+	/* breakpoint ratio: core floor 1.0, pb 2.0 (cache floor :: pb = 2/8) */
+	else if ((2 * freq_core_floor) >= (*fbc_freq)) {
+		*floor_ratio = FABRIC_CORE_FLOOR_RATIO_RATIO_2_8;
+	}
+	else {
+		printk (BIOS_ERR, "Unsupported core ceiling/PB frequency ratio = (%d/%d)",
+			freq_core_floor, *fbc_freq);
+		return -1;
+	}
+
+	/* breakpoint ratio: core ceiling 4.0, pb 2.0 (cache ceiling :: pb = 8/8) */
+	if (*freq_ceiling >= (2 * (*fbc_freq))) {
+		*ceiling_ratio = FABRIC_CORE_CEILING_RATIO_RATIO_8_8;
+	}
+	/* breakpoint ratio: core ceiling 3.5, pb 2.0 (cache ceiling :: pb = 7/8) */
+	else if ((4 * (*freq_ceiling)) >= (7 * (*fbc_freq))) {
+		*ceiling_ratio = FABRIC_CORE_CEILING_RATIO_RATIO_7_8;
+	}
+	/* breakpoint ratio: core ceiling 3.0, pb 2.0 (cache ceiling :: pb = 6/8) */
+	else if ((2 * (*freq_ceiling)) >= (3 * (*fbc_freq))) {
+		*ceiling_ratio = FABRIC_CORE_CEILING_RATIO_RATIO_6_8;
+	}
+	/* breakpoint ratio: core ceiling 2.5, pb 2.0 (cache ceiling :: pb = 5/8) */
+	else if ((4 * (*freq_ceiling)) >= (5 * (*fbc_freq))) {
+		*ceiling_ratio = FABRIC_CORE_CEILING_RATIO_RATIO_5_8;
+	}
+	/* breakpoint ratio: core ceiling 2.0, pb 2.0 (cache ceiling :: pb = 4/8) */
+	else if ((*freq_ceiling) >= (*fbc_freq)) {
+		*ceiling_ratio = FABRIC_CORE_CEILING_RATIO_RATIO_4_8;
+	}
+	/* breakpoint ratio: core ceiling 1.0, pb 2.0 (cache ceiling :: pb = 2/8) */
+	else if ((2 * (*freq_ceiling)) >= (*fbc_freq)) {
+		*ceiling_ratio = FABRIC_CORE_CEILING_RATIO_RATIO_2_8;
+	}
+	else {
+		printk (BIOS_ERR, "Unsupported core ceiling/PB frequency ratio = (%d/%d)",
+			*freq_ceiling, *fbc_freq);
+		return -1;
+	}
+
+	return 0;
+}
+
+static void config_guardband_epsilon(const uint8_t gb_percentage, uint32_t *target_value)
+{
+	uint32_t delta =  (*target_value * gb_percentage) / 100;
+	delta += ((*target_value * gb_percentage) % 100) ? 1 : 0;
+
+	printk(BIOS_DEBUG, "%s\n", __func__);
+
+	/* Apply guardband */
+	if (gb_percentage >= 0)
+	{
+		/* Clamp to maximum value if necessary */
+		if (delta > (EPSILON_MAX_VALUE - target_value)) {
+			printk(BIOS_DEBUG, "Guardband application generated out-of-range target value,"
+					   " clamping to maximum value!\n");
+			*target_value = EPSILON_MAX_VALUE;
+		} else {
+			*target_value += delta;
+		}
+	}
+	else
+	{
+		/* Clamp to minimum value if necessary */
+		if (delta >= *target_value) {
+			printk(BIOS_DEBUG, "Guardband application generated out-of-range target value,"
+					   " clamping to minimum value!\n");
+			*target_value = EPSILON_MIN_VALUE;
+		} else {
+			*target_value -= delta;
+		}
+	}
+
+}
+
+static void dump_eplisons(void)
```

Comment:
Typo, here and in invocations below
```suggestion
static void dump_epsilons(void)
```